server/http.go

```diff
@@ -54,26 +65,15 @@ func (p *Plugin) connectUserToZoom(w http.ResponseWriter, r *http.Request) {
 		return
 	}
 
-	channelID := r.URL.Query().Get("channelID")
-	if channelID == "" {
-		http.Error(w, "channelID missing", http.StatusBadRequest)
-		return
-	}
-
-	conf, err := p.getOAuthConfig()
-	if err != nil {
-		http.Error(w, err.Error(), http.StatusInternalServerError)
-	}
-
-	key := fmt.Sprintf("%v_%v", model.NewId()[0:15], userID)
-	state := fmt.Sprintf("%v_%v", key, channelID)
-
-	appErr := p.API.KVSet(key, []byte(state))
+	// fetch OAuth user state from the KV store that has been saved in '/zoom start' command handler
+	state, appErr := p.fetchOAuthUserState(userID)
```

Comment:
Fetching the state from KV store because we cannot re-build it ourselves here, as it contains a randomly-generated prefix and the channel ID.